data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -210,6 +221,72 @@ impl TraceExporter {
         TraceExporterBuilder::default()
     }
 
+    fn runtime(&self) -> Result<Arc<Runtime>, TraceExporterError> {
+        match self.runtime.lock_or_panic().as_ref() {
+            Some(runtime) => Ok(runtime.clone()),
+            None => self.run_worker(),
+        }
+    }
+
+    pub fn run_worker(&self) -> Result<Arc<Runtime>, TraceExporterError> {
+        let mut runtime_guard = self.runtime.lock_or_panic();
+        let runtime = match runtime_guard.as_ref() {
+            Some(runtime) => {
+                // Runtime already running
+                runtime.clone()
```

Comment:
/question

is it safe to clone `runtime`? or you are only cloning the reference?

Reply:
`runtime` is just and `Arc`, the clone is cheap.